LoadILLDiffractionTest assumes ILL facility

Similar to #21508, `LoadILLDiffractionTest` fails to find the data if the facility is not ILL. The test looks like it should be doing the right thing, but it doesn't work with `SNS` as the default facility (and `PG3` as default instrument) in the user properties file.
```
In LoadILLDiffractionTest::test_D20_multifile:
.../Framework/DataHandling/test/LoadILLDiffractionTest.h:225: Error: Test failed: Invalid value for property Filename (list of str lists) "967100-967101.nxs": When setting value of property "Slave": File "PG3_967100.nxs" not found
.../Framework/DataHandling/test/LoadILLDiffractionTest.h:225: Error: Expected (alg.setPropertyValue("Filename", "967100-967101.nxs")) not to throw, but it did
Load-[Notice] Load started
Load-[Information] Algorithm: Load v1
Load-[Information] Execution Date: 2019-01-03 15:50:43.313576000
Load-[Information] Execution Duration: -1 seconds
Load-[Information] UUID: 6193593a-fa52-42e1-8fd5-e89d0c627aed
Load-[Information] Parameters:
Load-[Information]   Name: Filename, Value: , Default?: Yes, Direction: Input
Load-[Information]   Name: OutputWorkspace, Value: _outWS, Default?: No, Direction: Output
Load-[Information]   Name: LoaderName, Value: , Default?: Yes, Direction: Output
Load-[Information]   Name: LoaderVersion, Value: -1, Default?: Yes, Direction: Output
PropertyManager-[Error] Property "FILENAME" is not set to a valid value: "File "" not found".
PropertyManager-[Error] Property "FILENAME" is not set to a valid value: "File "" not found".
.../Framework/DataHandling/test/LoadILLDiffractionTest.h:227: Error: Test failed: Some invalid Properties found
.../Framework/DataHandling/test/LoadILLDiffractionTest.h:227: Error: Expected (alg.execute()) not to throw, but it did
.../Framework/DataHandling/test/LoadILLDiffractionTest.h:228: Error: Assertion failed: alg.isExecuted()
```

--- a/Framework/DataHandling/test/LoadILLDiffractionTest.h
+++ b/Framework/DataHandling/test/LoadILLDiffractionTest.h
@@ -15,6 +15,7 @@
 #include "MantidDataHandling/LoadILLDiffraction.h"
 #include "MantidGeometry/Instrument/DetectorInfo.h"
 #include "MantidKernel/ConfigService.h"
+#include "MantidKernel/FacilityInfo.h"
 
 using namespace Mantid::API;
 using namespace Mantid::Kernel;
@@ -32,7 +33,22 @@ public:
   void setUp() override {
     ConfigService::Instance().appendDataSearchSubDir("ILL/D20/");
     ConfigService::Instance().appendDataSearchSubDir("ILL/D2B/");
+
+    m_oldFacility = ConfigService::Instance().getFacility().name();
     ConfigService::Instance().setFacility("ILL");
+
+    m_oldInstrument = ConfigService::Instance().getInstrument().name();
+    ConfigService::Instance().setString("default.instrument", "");
+  }
+
+  void tearDown() override {
+    if (!m_oldFacility.empty()) {
+      ConfigService::Instance().setFacility(m_oldFacility);
+    }
+    if (!m_oldInstrument.empty()) {
+      ConfigService::Instance().setString("default.instrument",
+                                          m_oldInstrument);
+    }
   }
 
   void test_Init() {
@@ -400,6 +416,8 @@ public:
 
 private:
   const double RAD_2_DEG = 180.0 / M_PI;
+  std::string m_oldFacility;
+  std::string m_oldInstrument;
 };
 
 #endif /* MANTID_DATAHANDLING_LOADILLDIFFRACTIONTEST_H_ */

